pkg/api/addons.go

```diff
@@ -522,67 +522,83 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		Containers: []KubernetesContainerSpec{
 			{
 				Name:           common.CSIProvisionerContainerName,
-				Image:          specConfig.MCRKubernetesImageBase + k8sComponents[common.CSIProvisionerContainerName],
+				Image:          specConfig.MCRKubernetesImageBase + getCSISidecarComponent(common.AzureDiskCSIDriverAddonName, common.CSIProvisionerContainerName, k8sComponents),
 				CPURequests:    "10m",
 				MemoryRequests: "20Mi",
-				CPULimits:      "200m",
-				MemoryLimits:   "200Mi",
+				CPULimits:      "2",
```

Comment:
The delta between the requests and limits is really broad here, do we feel O.K. about that?

Reply:
@andyzhangx could you comment on this? We increased the limits to prevent containers from restarting.

Reply:
200Mi is not enough, so we set an aggressive limit

Reply:
Then should we increase the CPURequests to 1? If we know that we will need up to 2 cores, shouldn't we reserve half to start to prevent the scheduler from thrashing?

Reply:
actually we don't know now, and usually windows pod would consume more cpu, memory than linux pod. This value is an aggressive value, need to do some perf test in the future. And if we set as half, we may waste resource. Shall we go with this current value now since it's still alpha/beta now?

Reply:
I agree with @jackfrancis. @andyzhangx what do you think of the new requests and limits in https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/363? We should keep the limits for Linux pods the same and slightly decrease resource requests and limits for Windows pods.

Reply:
For Linux:
```yaml
resources:
  limits:
    cpu: 200m
    memory: 200Mi
  requests:
    cpu: 10m
    memory: 20Mi
```

For Windows:
```yaml
resources:
  limits:
    cpu: 1
    memory: 1Gi
  requests:
    cpu: 500m
    memory: 512Mi
```

Reply:
what about this?
For Linux:
```yaml
resources:
  limits:
    cpu: 500m
    memory: 500Mi
  requests:
    cpu: 10m
    memory: 20Mi
```

For Windows:
```yaml
resources:
  limits:
    cpu: 1
    memory: 1Gi
  requests:
    cpu: 10m
    memory: 20Mi
```